.github/workflows/swift_package_test.yml

```diff
@@ -529,6 +528,26 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v1
         if: ${{ matrix.os_version == 'amazonlinux2' }}
+      - name: Checkout swiftlang/github-workflows repository
+        if: ${{ matrix.os_version != 'amazonlinux2' && github.repository != 'swiftlang/github-workflows' }}
+        uses: actions/checkout@v4
+        with:
+          repository: swiftlang/github-workflows
+          path: github-workflows
+      - name: Checkout swiftlang/github-workflows repository
+        if: ${{ matrix.os_version == 'amazonlinux2' && github.repository != 'swiftlang/github-workflows' }}
+        uses: actions/checkout@v1
```

Comment:
could we pin this to a specific revision, like `main`?
```ref: 'main' ```

I don't want to run the risk of running the workflow in this repository (say as part of a CI test) and having this check out a users branch which could contain a malicious version of the script being executed

Reply:
After re-reading the security policy about approving workflow runs
https://docs.github.com/en/actions/how-tos/manage-workflow-runs/approve-runs-from-forks

This might not be a real concern, since we have to approve all PR workflow runs from non-write-access-contributors.

 I'll defer to @shahmishal on this one

Reply:
'ref' defaults to the default branch of the repository, so I think this would not do anything interesting. Furthermore, someone could just put the malicious script inline in the YAML instead of referencing an external one by URL. You're right that workflow approvals are the real security tool here.